src/Model/Index/PathRepository.php

```diff
@@ -32,23 +32,24 @@ public function __construct(PathResource $resource, CollectionFactory $collectio
         $this->collectionFactory = $collectionFactory;
     }
 
-    public function getByPath(string $path): PathInterface
+    public function getByPath(string $path, string $storeCode = 'default'): PathInterface
     {
-        $model = $this->pathFactory->create();
-
-        $this->resource->load($model, $path, 'path');
+        $collection = $this->collectionFactory->create()
+            ->addFieldToFilter('path', $path)
```

Comment:
@Demoli This felt wrong doing this here but didn't see another way of testing

Reply:
What was the issue with the old method?

Reply:
Sorry I see now, that's fine, if you have multiple params to load by then a collection is acceptable.